core/msg.c

```diff
@@ -201,14 +210,25 @@ int msg_send_int(msg_t *m, kernel_pid_t target_pid)
         /* copy msg to target */
         msg_t *target_message = (msg_t*) target->wait_data;
         *target_message = *m;
+        target->flags |= THREAD_FLAG_MSG_WAITING;
         sched_set_status(target, STATUS_PENDING);
 
         sched_context_switch_request = 1;
         return 1;
     }
     else {
         DEBUG("msg_send_int: Receiver not waiting.\n");
+#if MODULE_CORE_THREAD_FLAGS
+        if (queue_msg(target, m)) {
+            thread_flags_wake(target);
+            return 1;
+        }
+        else {
+            return -1;
```

Comment:
if the receiver is not waiting, then you should probably return `0` and not `-1` (https://github.com/RIOT-OS/RIOT/blob/master/core/include/msg.h#L270)